### PR TITLE
UCP/PROTO: Extend protocol init params

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucp/proto/proto.h [hunk 1]
```diff
@@ -42,6 +42,16 @@ typedef unsigned ucp_proto_id_t;
 typedef uint64_t ucp_proto_id_mask_t;
 
 
+/**
+ * Protocol tag for internal usage, to allow searching for specific protocols
+ */
+typedef enum {
+    UCP_PROTO_TAG_NONE = 0,
```
Comment (dmitrygx) [inline src/ucp/proto/proto.h:49]: AFAIK, we rely in other places that an enumerator elements starting from `0`

File: src/ucp/proto/proto.h [hunk 2]
```diff
@@ -62,8 +72,11 @@ typedef struct {
  * Protocol and its private configuration
  */
 typedef struct {
-    const ucp_proto_t       *proto;  /* Protocol definition */
-    const void              *priv;   /* Protocol private configuration space */
+    const ucp_proto_t        *proto;    /* Protocol definition */
+    const void               *priv;     /* Protocol private configuration space */
+    ucp_proto_select_param_t sel_param; /* Copy of protocol selection parameters,
```
Comment (dmitrygx) [inline src/ucp/proto/proto.h:77]: maybe `select_*`?

File: src/ucp/proto/proto.h [hunk 3]
```diff
@@ -96,9 +109,13 @@ typedef struct {
  */
 typedef struct {
     /* Input parameters */
-    ucp_worker_h                   worker;         /* Worker to initialize on */
-    const ucp_proto_select_param_t *sel_param;     /* Operation parameters */
-    const ucp_ep_config_key_t      *ep_config_key; /* Endpoint configuration */
+    ucp_worker_h                   worker;           /* Worker to initialize on */
+    const ucp_proto_select_param_t *sel_param;       /* Operation parameters */
```
Comment (dmitrygx) [inline src/ucp/proto/proto.h:113]: maybe `select_*`?

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
#4616
bot:pipe:retest

**Comment by brminich** [comment]
issue is #5461 

bot:pipe:retest

File: src/ucp/proto/proto.h [hunk 4]
```diff
@@ -42,6 +42,16 @@ typedef unsigned ucp_proto_id_t;
...
 
 
+/**
+ * Protocol tag for internal usage, to allow searching for specific protocols
+ */
+typedef enum {
+    UCP_PROTO_TAG_NONE,
+    UCP_PROTO_TAG_AM_SHORT,   /* The protocol uses only uct_ep_am_short() */
+    UCP_PROTO_TAG_PUT_SHORT   /* The protocol uses only uct_ep_put_short() */
+} ucp_proto_tag_t;
```
Comment (brminich) [inline src/ucp/proto/proto.h:52]: id' not use `tag` in the name - it is confusing, like it is related to TAG API only.
maybe `ucp_proto_id_t`?

Comment (yosefe) [reply src/ucp/proto/proto.h:52]: yes, agree
id is also not suitable because it does not have to be unique
maybe "proto_flag_t" ?

Comment (brminich) [reply src/ucp/proto/proto.h:52]: do you mean it can be a combination of, say, `BCOPY` and `MULTI`?
Then `proto_flag_t` is much better name.

Comment (yosefe) [reply src/ucp/proto/proto.h:52]: yes, make this a combination instead of enum, probably can be useful in the future

File: src/ucp/proto/proto.h [hunk 5]
```diff
@@ -42,6 +42,15 @@ typedef unsigned ucp_proto_id_t;
 typedef uint64_t ucp_proto_id_mask_t;
 
 
+/**
+ * Protocol flags for internal usage, to allow searching for specific protocols
+ */
+enum {
+    UCP_PROTO_FLAG_AM_SHORT  = UCS_BIT(0), /* The protocol uses only uct_ep_am_short() */
+    UCP_PROTO_FLAG_PUT_SHORT = UCS_BIT(0)  /* The protocol uses only uct_ep_put_short() */
```
Comment (brminich) [inline src/ucp/proto/proto.h:50]: the value is the same as `UCP_PROTO_FLAG_AM_SHORT` intentionally, right?
So there will be names for every API which support certain protocol?

Comment (yosefe) [reply src/ucp/proto/proto.h]: sorry that was a typo, fixed

**Pull request merged**